asf_heat_pump_affordability/notebooks/identify_and_review_analytical_sample.py

```diff
@@ -0,0 +1,338 @@
+
+exclusion_criteria.append("drop_row")
+
+plot_counts = {"labels": exclusion_criteria}
+plot_counts["counts"] = [df[ec].sum() for ec in exclusion_criteria]
+plot_df = pd.DataFrame(plot_counts).sort_values("counts")
+
+fig, ax = plt.subplots()
+bar = ax.bar(x=plot_df["labels"], height=plot_df["counts"])
+ax.bar_label(bar)
+ax.set_ylim([0, round(max(plot_df["counts"]) + 1000, -3)])
+ax.set_title(
+    "Number of rows removed for individual exclusion criteria\nand net exclusion effect"
+)
+plt.xticks(rotation=90)
+plt.show()
+
+# %%
+## See count and proportion of NaN values for each key variable
+## See count and proportion of non-NaN values lost for each key variable when dropping rows based on exclusion criteria
+
+_cols = [
+    "cost_na",
+    "tech_type_na",
+    "original_epc_index_na",
+    "property_archetype_data_na",
+]
+lost_vals = {
+    k: []
+    for k in [
+        "variable",
+        "na_count",
+        "proportion_na",
+        "non_na_lost_count",
+        "non_na_proportion_lost",
+    ]
+}
+
+for _col in _cols:
+    lost_count = df[(df[_col] == False) & (df["drop_row"] == True)]["drop_row"].sum()
+    lost_vals["variable"].append(_col[:-3])
+    lost_vals["na_count"].append(df[_col].sum())
+    lost_vals["proportion_na"].append(df[_col].sum() / len(df))
+    lost_vals["non_na_lost_count"].append(lost_count)
+    lost_vals["non_na_proportion_lost"].append(lost_count / len(df[df[_col] == False]))
+
+pd.DataFrame(lost_vals)
+
+# %%
+# print(f"Total rows dropped: {(df['drop_row'] == True).sum()}")
+# print(f"Total proportion of rows dropped: {(df['drop_row'] == True).sum() / len(df)}")
+
+print(f"Total rows dropped: {df['drop_row'].sum()}")
+print(f"Total proportion of rows dropped: {df['drop_row'].sum() / len(df)}")
+
+# %%
+## Keep rows where drop_row is False, i.e. remove rows based on exclusion criteria
+# sample = df[df["drop_row"] == False]
+sample = df[~df["drop_row"]]
+
+# %%
+## Conduct checks to ensure appropriate rows dropped
+
+for _col in ["cost", "tech_type", "original_epc_index"]:
+    assert sample[_col].isna().sum() == 0
+
+for _col in [
+    "cost_na",
+    "tech_type_na",
+    "original_epc_index_na",
+    "property_archetype_data_na",
+]:
+    assert sample[_col].unique() == False
+
+assert sample["tech_type"].unique() == "Air Source Heat Pump"
+
+# %% [markdown]
+# ## See distribution of removed cost data
+
+# %% [markdown]
+# Removed rows either have no EPC data or are not an ASHP.
+# We will see if the distribution of costs in removed rows is the same as that of rows which have been kept in the sample for analysis.
+
+# %%
+dropped_rows = filter_by_year.loc[~filter_by_year.index.isin(sample.index)]
+assert (len(filter_by_year) - len(sample)) == len(dropped_rows)
+
+# %%
+# In this case we'll normalise the data so they're comparable.
+
+fig, ax = plt.subplots(nrows=1, ncols=2, sharey=True, figsize=(10, 3))
+ax[0].hist(sample["cost"], density=True)
```

Comment:
I originally left the `density` arg out as I believe it makes the area under the graph integrate to 1 so the y values became very small, which ultimately meant I didn't have a good intuition of how to interpret the y-axis and wasn't sure about its readability in general. 

I understand the reasoning to normalise the data so they are comparable across the two samples and I guess the important thing is to compare the shapes of the distributions so the y-axis is less relevant (?). Can you input on how to interpret the y-axis?

Reply:
The y axis is the density as you've stated. It's the absolute value relative to the sum of absolute values for the distribution. You can think of it as a scaling. It has no real interpretation, it just makes distributions with different volumes comparable while maintaining their shape.